crates/apollo-compiler/tests/validation/types.rs

```diff
@@ -0,0 +1,1290 @@
+//! graphql-js they do.
+use std::sync::OnceLock;
+
+use apollo_compiler::ExecutableDocument;
+use apollo_compiler::Schema;
+
+const GRAPHQL_JS_TEST_SCHEMA: &str = r#"
+  interface Mammal {
+    mother: Mammal
+    father: Mammal
+  }
+
+  interface Pet {
+    name(surname: Boolean): String
+  }
+
+  interface Canine implements Mammal {
+    name(surname: Boolean): String
+    mother: Canine
+    father: Canine
+  }
+
+  enum DogCommand {
+    SIT
+    HEEL
+    DOWN
+  }
+
+  type Dog implements Pet & Mammal & Canine {
+    name(surname: Boolean): String
+    nickname: String
+    barkVolume: Int
+    barks: Boolean
+    doesKnowCommand(dogCommand: DogCommand): Boolean
+    isHouseTrained(atOtherHomes: Boolean = true): Boolean
+    isAtLocation(x: Int, y: Int): Boolean
+    mother: Dog
+    father: Dog
+  }
+
+  type Cat implements Pet {
+    name(surname: Boolean): String
+    nickname: String
+    meows: Boolean
+    meowsVolume: Int
+    furColor: FurColor
+  }
+
+  union CatOrDog = Cat | Dog
+
+  type Human {
+    name(surname: Boolean): String
+    pets: [Pet]
+    relatives: [Human]!
+  }
+
+  enum FurColor {
+    BROWN
+    BLACK
+    TAN
+    SPOTTED
+    NO_FUR
+    UNKNOWN
+  }
+
+  input ComplexInput {
+    requiredField: Boolean!
+    nonNullField: Boolean! = false
+    intField: Int
+    stringField: String
+    booleanField: Boolean
+    stringListField: [String]
+  }
+
+  # TODO oneOf not supported in apollo-rs
+  input OneOfInput { # @oneOf
+    stringField: String
+    intField: Int
+  }
+
+  type ComplicatedArgs {
+    # TODO List
+    # TODO Coercion
+    # TODO NotNulls
+    intArgField(intArg: Int): String
+    nonNullIntArgField(nonNullIntArg: Int!): String
+    stringArgField(stringArg: String): String
+    booleanArgField(booleanArg: Boolean): String
+    enumArgField(enumArg: FurColor): String
+    floatArgField(floatArg: Float): String
+    idArgField(idArg: ID): String
+    stringListArgField(stringListArg: [String]): String
+    stringListNonNullArgField(stringListNonNullArg: [String!]): String
+    complexArgField(complexArg: ComplexInput): String
+    oneOfArgField(oneOfArg: OneOfInput): String
+    multipleReqs(req1: Int!, req2: Int!): String
+    nonNullFieldWithDefault(arg: Int! = 0): String
+    multipleOpts(opt1: Int = 0, opt2: Int = 0): String
+    multipleOptAndReq(req1: Int!, req2: Int!, opt1: Int = 0, opt2: Int = 0): String
+  }
+
+  type QueryRoot {
+    human(id: ID): Human
+    dog: Dog
+    cat: Cat
+    pet: Pet
+    catOrDog: CatOrDog
+    complicatedArgs: ComplicatedArgs
+  }
+
+  schema {
+    query: QueryRoot
+  }
+
+  directive @onField on FIELD
+"#;
+
+fn test_schema() -> &'static Schema {
+    static SCHEMA: OnceLock<Schema> = OnceLock::new();
+
+    SCHEMA.get_or_init(|| {
+        let schema = Schema::parse(GRAPHQL_JS_TEST_SCHEMA, "schema.graphql");
+        schema.validate().unwrap();
+        schema
+    })
+}
+
+fn expect_valid(query: &'static str) {
```

Comment:
```suggestion
#[track_caller]
fn expect_valid(query: &'static str) {
```

Only relevant if a test starts failing, but panics pointing to the caller would be helpful. Though maybe the test name is enough